src/Parser.php

```diff
@@ -2453,10 +2457,10 @@ private function parsePropertyDeclaration($parentNode, $modifiers) {
         return $propertyDeclaration;
     }
 
-    private function parseQualifiedNameList($parentNode) {
+    private function parseQualifiedNameList($parentNode, $delimiter = TokenKind::CommaToken) {
         return $this->parseDelimitedList(
             DelimitedList\QualifiedNameList::class,
-            TokenKind::CommaToken,
+            $delimiter,
```

Comment:
Oh, I see what you meant in the delimited list PR - yeah, I don't think this is a super common case to handle, so I think it makes sense to special case this usage (or come up with some other idea that doesn't add additional memory usage.)